From 3fe2fb49c613641da78ac1d176415387af1573ff Mon Sep 17 00:00:00 2001
From: Mladen Milinkovic <maxrd2@smoothware.net>
Date: Wed, 13 Oct 2021 23:55:01 +0200
Subject: Fixed crash when closing subtitle

---
 src/application_subtitle.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/application_subtitle.cpp b/src/application_subtitle.cpp
index 64714888..c56025c9 100644
--- a/src/application_subtitle.cpp
+++ b/src/application_subtitle.cpp
@@ -393,10 +393,10 @@ Application::closeSubtitle()
 			emit translationModeChanged(false);
 		}
 
-		m_undoStack->clear();
-
 		emit subtitleClosed();
 
+		m_undoStack->clear();
+
 		m_subtitle.reset();
 
 		m_labSubFormat->setText(QString());
-- 
2.38.1

